src/main/scala/io/iohk/ethereum/consensus/FullConsensusConfig.scala

```diff
@@ -0,0 +1,31 @@
+package io.iohk.ethereum.consensus
+
+import io.iohk.ethereum.consensus.ethash.MiningConfig
+
+case class FullConsensusConfig[C <: AnyRef /*Product*/](
+  generic: ConsensusConfig,
+  specific: C
+) {
+  // Sanity check
+  require(
+    generic.protocol.isEthash == ifEthash(_ ⇒ true)(false),
+    "generic.protocol.isEthash == ifEthash(_ ⇒ true)(false)"
+  )
+
+  /**
+   * There are APIs that expect that the standard Ethash consensus is running and so depend
+   * on either its configuration or general PoW semantics.
+   * This is a method that can handle such cases via a respective if/then/else construct:
+   * if we run under [[io.iohk.ethereum.consensus.Ethash Ethash]] the `_if` function is called,
+   * otherwise the `_else` value is computed.
+   */
+  final def ifEthash[A](_if: MiningConfig ⇒ A)(_else: ⇒ A): A =
```

Comment:
Why not just implement the method `isEthash` (`is` not `if`) in `EthService`?

Reply:
:+1: If this check (ethash vs non-ethash) is just for RPC it should live there.

Reply:
This check is currently implemented via a reflective pattern match that locally reveals the actual config implementation. I would like to keep it near where config implementations are (definition site) rather than near the use site.

In an alternative implementation strategy, e.g. using polymorphism so that only the ethash-related implementation knows how to use the `_if` branch (because it's the only one with a `MiningConfig` instance), the above remark is strengthened even more.

Also `isEthash` alone is not enough, we need the exact `MiningConfig` instance. My [other comment](https://github.com/input-output-hk/mantis/pull/407#discussion_r171520270) is also relevant.